NebulaModel/Packets/Universe/DysonSphereResponse.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace NebulaModel.Packets.Universe
+{
+    class DysonSphereResponse
+    {
+        public int[] StarIndexes { get; set; }
+        public byte[] DysonSphereBytes { get; set; }
+        public int[] DysonSphereBytesLengths { get; set; }
+
+        public DysonSphereResponse() { }
+
+        public DysonSphereResponse(Dictionary<int, byte[]> dysonSphereData)
+        {
+            this.StarIndexes = dysonSphereData.Keys.ToArray();
+
+            // Can't use a jagged array because LNL serializer says no, so flattening and separate offset array it is
+            // TODO: Possibly register a type with the serializer instead of doing this manually
+            this.DysonSphereBytes = dysonSphereData.Values.SelectMany(x => x).ToArray();
```

Comment:
Isn't this the biggest chunk of the memory used by the save file? Allocating arrays here feels really heavy.

Reply:
True, the big full Sphere can have around 9MB and a small full sphere around 5MB. I was following [PlanetDataResponse](https://github.com/hubastard/nebula/blob/0f1ba59d099e9d3b908fcf9c78c25ed982aed3bd/NebulaModel/Packets/Planet/PlanetDataResponse.cs#L21). But I can try to optimize it.

Reply:
Oh, I am not even using this class anymore, so I will check it and remove it :)

Reply:
removed.